crates/wallet/src/wallet/mod.rs

```diff
@@ -1557,6 +1556,19 @@ impl Wallet {
         params.ordering.sort_tx_with_aux_rand(&mut tx, rng);
 
         let psbt = self.complete_transaction(tx, coin_selection.selected, params)?;
+
+        if psbt
+            .unsigned_tx
+            .output
+            .iter()
+            .any(|txout| txout.script_pubkey == drain_script)
+        {
```

Comment:
It will be more *correct* to check the `excess` variable. The caller may set the recipient address as the drain address that we obtained from the index. In this case, it might be unexpected for the caller to have this spk marked as used.

```rust
        if matches!(excess, &coin_selection::Excess::Change { .. }) { }
```

Reply:
Yes that will be better, I think it's resolved by 6afde0b7761113775bdca11ee720777cd178acf7